dart/test/sentry_client_test.dart

```diff
@@ -1040,7 +1053,8 @@ void main() {
       expect(capturedEvent.user?.email, fakeEvent.user!.email);
     });
 
-    test('event has a user without IP address', () async {
+    test('event has a user without IP address and sendDefaultPii = true',
+        () async {
       final client = fixture.getSut(sendDefaultPii: true);
 
       final event = fakeEvent.copyWith(user: fakeUser);
```

Comment:
In the test `event has a user with IP address` we assume the user has an ip-address. Here we assume the user has none, but we do not set it to null anywhere in the test. We need to set it to null before calling captureEvent.

<img width="1325" alt="Bildschirmfoto 2025-02-19 um 14 32 58" src="https://github.com/user-attachments/assets/5ead1dd2-f239-4abf-8a6d-5e709b0a2b32" />

Reply:
you need to check expect(event.user!.ipAddress, isNull);

Reply:
actually let me add that assertion to be safe